pkg/apis/core/validation/utils.go

```diff
@@ -195,3 +195,39 @@ func ValidateIPFamilies(ipFamilies []core.IPFamily, fldPath *field.Path) field.E
 
 	return allErrs
 }
+
+func validateDependencyWatchdogSettings(settings *core.SeedSettingDependencyWatchdog, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if settings.Endpoint != nil && settings.Weeder != nil {
+		allErrs = append(allErrs, field.Forbidden(fldPath, "only one of spec.settings.dependencyWatchdog.weeder and spec.settings.dependencyWatchdog.endpoint can be specified, not both."))
+	}
+	if settings.Probe != nil && settings.Prober != nil {
+		allErrs = append(allErrs, field.Forbidden(fldPath, "only one of spec.settings.dependencyWatchdog.prober and spec.settings.dependencyWatchdog.probe can be specified, not both."))
+
+	}
+
+	return allErrs
+}
```

Comment:
You could add it here, https://github.com/gardener/gardener/blob/2f34cb1422985253aa47eb761bb759d6ae4823e3/pkg/registry/core/seed/strategy.go#L71-L75 so that it's not called on update.